packages/vue/src/sdk.ts

```diff
@@ -0,0 +1,414 @@
+          vm.$options[internalHook] = [handler, currentValue];
+        } else {
+          vm.$options[internalHook] = [handler];
+        }
+      });
+    });
+  };
+
+  /** Finish top-level span and activity with a debounce configured using `timeout` option */
+  private _finishRootSpan(timestamp: number): void {
+    if (this._rootSpanTimer) {
+      clearTimeout(this._rootSpanTimer);
+    }
+
+    this._rootSpanTimer = setTimeout(() => {
+      // We should always finish the span, only should pop activity if using @sentry/apm
+      if (this._rootSpan) {
+        this._rootSpan.finish(timestamp);
+        this._rootSpan = undefined;
+      }
+    }, this._options.tracingOptions.timeout);
+  }
+
+  /** Inject configured tracing hooks into Vue's component lifecycles */
+  private _startTracing(): void {
+    const applyTracingHooks = this._applyTracingHooks;
+    const appliedTracingHooks = setTimeout(() => {
+      logger.warn("Didn't apply tracing hooks, make sure you call Sentry.init before initialzing Vue!");
+    }, 500);
+    this._options.Vue.mixin({
+      beforeCreate(this: ViewModel): void {
+        clearTimeout(appliedTracingHooks);
+        applyTracingHooks(this);
+      },
+    });
+  }
+
+  /** Inject Sentry's handler into owns Vue's error handler  */
+  private _attachErrorHandler(): void {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    const currentErrorHandler = this._options.Vue.config.errorHandler;
+
+    this._options.Vue.config.errorHandler = (error: Error, vm?: ViewModel, info?: string): void => {
+      const metadata: Metadata = {};
+
+      if (vm) {
+        try {
+          metadata.componentName = this._getComponentName(vm);
+
+          if (this._options.attachProps) {
+            metadata.propsData = vm.$options.propsData;
+          }
+        } catch (_oO) {
+          logger.warn('Unable to extract metadata from Vue component.');
+        }
+      }
+
+      if (info) {
+        metadata.lifecycleHook = info;
+      }
+
+      // Capture exception in the next event loop, to make sure that all breadcrumbs are recorded in time.
+      setTimeout(() => {
+        getCurrentHub().withScope(scope => {
+          scope.setContext('vue', metadata);
+          getCurrentHub().captureException(error);
+        });
+      });
+
+      if (typeof currentErrorHandler === 'function') {
+        currentErrorHandler.call(this._options.Vue, error, vm, info);
+      }
+
+      if (this._options.logErrors) {
+        if (this._options.Vue.util) {
+          this._options.Vue.util.warn(`Error in ${info}: "${error.toString()}"`, vm);
+        }
+        // eslint-disable-next-line no-console
+        console.error(error);
+      }
+    };
+  }
+}
+
+/** Grabs active transaction off scope, if any */
+export function getActiveTransaction<T extends Transaction>(hub: Hub = getCurrentHub()): T | undefined {
+  return hub?.getScope()?.getTransaction() as T | undefined;
+}
```

Comment:
```suggestion
export function getActiveTransaction(hub: Hub = getCurrentHub()): Transaction | undefined {
  return hub?.getScope()?.getTransaction();
}
```

We never overload the generic, so it should be completely fine to just do that instead.

Same goes for `VueRouterInstrumentation` type.

Reply:
We can also change it directly to `getCurrentHub()` call, as the argument is always skipped in all the invocations.